src/utils/proxy.ts

```diff
@@ -691,9 +703,11 @@ const initializeProxy = async function ({
 
       if (htmlInjections) {
         responseBody = await injectHtml(responseBody, proxyRes, htmlInjections)
-        proxyResHeaders = {
-          ...proxyResHeaders,
-          'content-length': String(responseBody.byteLength),
+        if (!proxyResHeaders['transfer-encoding']) {
```

Comment:
@smnh you added this, but when there is transfer-encoding, you should not send content-length.
https://github.com/sindresorhus/got/discussions/1576

WDYT about this change?

Reply:
According to MDN, It looks like it should be ommitted only when `Transfer-Encoding` is `chunked`.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding#syntax

And it make sense, because for regular non chunked encodings like gzip and deflate you do want to have content-length.

```
if (proxyResHeaders['transfer-encoding']?.includes('chunked') {
```